server/embed/config.go

```diff
@@ -941,6 +941,13 @@ func (cfg *configYAML) configFromFile(path string) error {
 		cfg.ClusterState = ClusterStateFlagNew
 	}
 
+	if cfg.ExperimentalCompactionBatchLimit != 0 {
+		if cfg.CompactionBatchLimit != 0 {
+			return fmt.Errorf(`cannot set "experimentental-compaction-batch-limit" and "compaction-batch-limit" at the same time, please use "compaction-batch-limit" only`)
+		}
```

Comment:
It's already been covered below,

https://github.com/etcd-io/etcd/blob/9da01a8275c4b8e1b3838ef92cf74d00e41b2edf/server/embed/config.go#L1023-L1026

Reply:
This is for flags, not for JSON config.